src/server.cpp

```diff
@@ -3707,7 +3707,10 @@ bool Server::dynamicAddMedia(const DynamicMediaArgs &a)
 	// Push file to existing clients
 	if (m_env) {
 		NetworkPacket pkt(TOCLIENT_MEDIA_PUSH, 0);
-		pkt << raw_hash << filename << static_cast<bool>(a.ephemeral);
+		pkt << raw_hash << filename << static_cast<bool>(!a.ephemeral);
+		// NOTE: the meaning of a.ephemeral was accidentally inverted between proto 39 and 40,
+		// when dynamic_add_media v2 was added. As of 5.12.0 the server sends it correctly again.
+		// Compatibility code on the client-side was not added.
```

Comment:
```suggestion
		pkt << raw_hash << filename;
		// NOTE: the meaning of this field was accidentally inverted between proto 39 and 40,
		// when dynamic_add_media v2 was added. As of 5.12.0 the server sends it correctly again.
		// Compatibility code on the client-side was not added.
		pkt << static_cast<bool>(!a.ephemeral);
```

Reply:
There are several fields on the line above, so it is ambiguous whether "this field" is filename, or ephemeral, or something else.  Anybody reading this PR will know what it means, but somebody new to the code will have no idea.

Have you looked at #15886 ?  It has the advantage of working for new-client to old-server.  I cannot test whether it has a disadvantage in the old-client to new-server situation.

Reply:
Ooops - I read that wrong.  *sigh*  Must be getting tired.

Reply:
The idea was to split it to a different line so that "this field" is unambiguous.
If you disagree that's fine.